File: Trabalho3/app/app/src/main/java/br/ufc/quixada/dsdm/trabalho3/network/CarroService.java
package br.ufc.quixada.dsdm.trabalho3.network;

import br.ufc.quixada.dsdm.trabalho3.models.Carro;
import retrofit2.Call;
import retrofit2.http.*;

import java.util.List;

public interface CarroService {

    @GET("carros")
    Call<List<Carro>> list();

    @GET("carros/{id}")
    Call<Carro> retrieve(@Path("id") Integer id);

    @POST("carros")
    Call<Carro> register(@Body Carro carro);

    @PUT("carros/{id}")
    Call<Carro> update(@Path("id") Integer id, @Body Carro carro);

}

File: Trabalho3/server/services/CarroService.js
const { CarroModel } = require("../models/CarroModel");

let _id = 0;
let carros = [];

class CarroService {

    static list() {
        return carros;
    }

    static retrieve(id) {
        for (var carro of carros)
            if (carro.id == id)
                return carro;
        return null;
    }

    static register(data) {
        const carro = new CarroModel(
            _id++,
            data.modelo,
            data.marca,
            data.cor,
            data.anoDeLancamento,
        );

        carros.push(carro);
        return carro;
    }

    static update(id, data) {
        for (var carro of carros)
            if (carro.id == id) {
                carro.modelo = data.modelo;
                carro.marca = data.marca;
                carro.cor = data.cor;
                carro.anoDeLancamento = data.anoDeLancamento;
                return carro;
            }
        return null;
    }

}

module.exports = { CarroService };